lockfile/src/cargo.rs

```diff
@@ -0,0 +1,164 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+use serde::Deserialize;
+
+use crate::{Parse, ParseResult};
+
+pub struct Cargo;
+
+#[derive(Deserialize, Debug, Clone)]
+struct CargoLock {
+    #[serde(rename = "package")]
+    packages: Vec<Package>,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+struct Package {
+    name: String,
+    version: String,
+    source: Option<String>,
+}
+
+impl Parse for Cargo {
+    /// Parses `cargo.lock` files into a vec of packages
+    fn parse(&self, data: &str) -> ParseResult {
+        let mut lock: CargoLock = toml::from_str(data).unwrap();
+        Ok(lock
+            .packages
+            .drain(..)
+            .filter_map(|package| {
+                package.source.as_ref().map(|_| PackageDescriptor::from(package.clone()))
+            })
+            .collect())
+    }
+
+    fn package_type(&self) -> PackageType {
+        println!("declared package type");
+        PackageType::Cargo
+    }
+
+    fn is_path_lockfile(&self, path: &Path) -> bool {
+        path.file_name() == Some(OsStr::new("Cargo.lock"))
+    }
+}
+
+impl From<Package> for PackageDescriptor {
+    fn from(package: Package) -> Self {
+        let source = package.source.unwrap_or_default();
+        let version = if source.starts_with("git+") {
+            source.trim_start_matches("git+").into()
+        } else {
+            package.version
+        };
+
+        Self { name: package.name, package_type: PackageType::Cargo, version }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn parse_cli_cargo_lock() {
+        let pkgs = Cargo.parse(include_str!("../../tests/fixtures/Cargo.lock")).unwrap();
+        assert_eq!(pkgs.len(), 530);
+
+        let expected_pkgs = [
+            PackageDescriptor {
+                name: "Inflector".into(),
+                version: "0.11.4".into(),
+                package_type: PackageType::Cargo,
+            },
+            PackageDescriptor {
+                name: "adler".into(),
+                version: "1.0.2".into(),
+                package_type: PackageType::Cargo,
+            },
+            PackageDescriptor {
+                name: "aead".into(),
+                version: "0.5.1".into(),
+                package_type: PackageType::Cargo,
+            },
+            PackageDescriptor {
+                name: "aes".into(),
+                version: "0.8.1".into(),
+                package_type: PackageType::Cargo,
+            },
+            PackageDescriptor {
+                name: "landlock".into(),
+                version: "https://github.com/phylum-dev/rust-landlock#b553736cefc2a740eda746e5730cf250b069a4c1".into(),
+                package_type: PackageType::Cargo,
+            },
+        ];
+
+        for expected_pkg in expected_pkgs {
+            assert!(pkgs.contains(&expected_pkg));
+        }
+    }
+
+    #[test]
+    fn parse_cargo_lockv1() {
```

Comment:
This is just a very minor nitpick, but I'd separate the `v1` here with an underscore (and below, in the fixture names, with a dash):
```suggestion
    fn parse_cargo_lock_v1() {
```

Reply:
Thank you, it looks much cleaner with the underscore.